static/js/drafts.js

```diff
@@ -392,10 +411,40 @@ export function launch() {
         return sorted_formatted_drafts;
     }
 
-    function render_widgets(drafts) {
+    function render_widgets(narrow_drafts, other_drafts) {
         $("#drafts_table").empty();
+
+        let is_private = false;
+        let recipient;
+        // Compose box headers get priority.
+        if (compose_state.get_message_type() === "stream" && compose_state.topic()) {
+            recipient = `#${compose_state.stream_name()} > ${compose_state.topic()}`;
+        } else if (compose_state.get_message_type() === "stream") {
+            recipient = `#${compose_state.stream_name()}`;
+        } else if (compose_state.get_message_type() === "private") {
+            recipient = people.recipient_names_from_emails(
+                compose_state.private_message_recipient().split(","),
+            );
+            is_private = true;
+        } else if (narrow_state.narrowed_to_topic()) {
+            recipient = `#${narrow_state.stream()} > ${narrow_state.topic()}`;
+        } else if (narrow_state.stream()) {
+            recipient = `#${narrow_state.stream()}`;
+        } else if (narrow_state.narrowed_by_pm_reply()) {
+            recipient = people.recipient_names_from_emails(
+                narrow_state.pm_emails_string().split(","),
+            );
+            is_private = true;
+        }
```

Comment:
The block from here up could benefit from being an independent function that just returns `{recipient: ..., is_private: false|true}` in each block; the main thing you'd get out of that is removing all the if/else chaining, but it'd also be more naturally a thing you could test.

Reply:
I don't see much benefit to changing if/else chaining into a list of ifs, or is there somehow a way to avoid that? If I were to make a function, I'd have it return the full string `narrow_drafts_header` instead of the half-done calculation.

Reply:
Ooooo I can reuse the function I pulled out below and that works sooo much better yay!